<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>600</width>
    <height>450</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <widget class="QLabel" name="label">
    <property name="geometry">
     <rect>
      <x>100</x>
      <y>0</y>
      <width>400</width>
      <height>100</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>30</pointsize>
     </font>
    </property>
    <property name="text">
     <string>峰峰的字符串匹配器</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
   <widget class="QPushButton" name="buttonTextGenerate">
    <property name="geometry">
     <rect>
      <x>400</x>
      <y>95</y>
      <width>85</width>
      <height>32</height>
     </rect>
    </property>
    <property name="text">
     <string>Generate</string>
    </property>
   </widget>
   <widget class="QPushButton" name="buttonStart">
    <property name="geometry">
     <rect>
      <x>480</x>
      <y>400</y>
      <width>85</width>
      <height>32</height>
     </rect>
    </property>
    <property name="text">
     <string>Start!</string>
    </property>
   </widget>
   <widget class="QPushButton" name="buttonTextHelp">
    <property name="geometry">
     <rect>
      <x>480</x>
      <y>95</y>
      <width>85</width>
      <height>32</height>
     </rect>
    </property>
    <property name="text">
     <string>Help</string>
    </property>
   </widget>
   <widget class="QGroupBox" name="groupAlgorithm">
    <property name="geometry">
     <rect>
      <x>80</x>
      <y>310</y>
      <width>201</width>
      <height>91</height>
     </rect>
    </property>
    <property name="title">
     <string>匹配算法</string>
    </property>
    <widget class="QRadioButton" name="radioBF">
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>20</y>
       <width>201</width>
       <height>20</height>
      </rect>
     </property>
     <property name="text">
      <string>BF(Brute-Force)</string>
     </property>
     <property name="checked">
      <bool>true</bool>
     </property>
    </widget>
    <widget class="QRadioButton" name="radioKMP">
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>40</y>
       <width>201</width>
       <height>20</height>
      </rect>
     </property>
     <property name="text">
      <string>KMP(Knuth-Morris-Pratt)</string>
     </property>
    </widget>
    <widget class="QRadioButton" name="radioBM">
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>60</y>
       <width>201</width>
       <height>20</height>
      </rect>
     </property>
     <property name="text">
      <string>BM(Boyer-Moore)</string>
     </property>
    </widget>
   </widget>
   <widget class="QGroupBox" name="groupSigma">
    <property name="geometry">
     <rect>
      <x>80</x>
      <y>220</y>
      <width>201</width>
      <height>81</height>
     </rect>
    </property>
    <property name="title">
     <string>串字符集</string>
    </property>
    <widget class="QRadioButton" name="radioSigma0">
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>20</y>
       <width>201</width>
       <height>20</height>
      </rect>
     </property>
     <property name="text">
      <string>abc</string>
     </property>
     <property name="checked">
      <bool>true</bool>
     </property>
    </widget>
    <widget class="QRadioButton" name="radioSigma1">
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>40</y>
       <width>201</width>
       <height>20</height>
      </rect>
     </property>
     <property name="text">
      <string>0-9</string>
     </property>
    </widget>
    <widget class="QRadioButton" name="radioSigma2">
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>60</y>
       <width>201</width>
       <height>20</height>
      </rect>
     </property>
     <property name="text">
      <string>ASCII 33-126</string>
     </property>
    </widget>
   </widget>
   <widget class="QLabel" name="label_2">
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>100</y>
      <width>151</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>文本串：随机生成长度为</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="lineTextSize">
    <property name="geometry">
     <rect>
      <x>180</x>
      <y>100</y>
      <width>113</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>1000</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_3">
    <property name="geometry">
     <rect>
      <x>300</x>
      <y>100</y>
      <width>101</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>(≤10^6)的文本串</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_5">
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>180</y>
      <width>151</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>匹配串：随机生成长度为</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="lineMatchSize">
    <property name="geometry">
     <rect>
      <x>180</x>
      <y>180</y>
      <width>113</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>4</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="lineMatchString">
    <property name="geometry">
     <rect>
      <x>80</x>
      <y>140</y>
      <width>321</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>acab</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_6">
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>140</y>
      <width>51</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>匹配串：</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_4">
    <property name="geometry">
     <rect>
      <x>300</x>
      <y>180</y>
      <width>101</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>(≤10^6)的匹配串</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_7">
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>220</y>
      <width>41</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>选项：</string>
    </property>
   </widget>
   <widget class="QPushButton" name="buttonMatchHelp">
    <property name="geometry">
     <rect>
      <x>480</x>
      <y>175</y>
      <width>85</width>
      <height>32</height>
     </rect>
    </property>
    <property name="text">
     <string>Help</string>
    </property>
   </widget>
   <widget class="QPushButton" name="buttonMatchGenerate">
    <property name="geometry">
     <rect>
      <x>400</x>
      <y>175</y>
      <width>85</width>
      <height>32</height>
     </rect>
    </property>
    <property name="text">
     <string>Generate</string>
    </property>
   </widget>
   <widget class="QPushButton" name="buttonSigmaHelp">
    <property name="geometry">
     <rect>
      <x>480</x>
      <y>230</y>
      <width>85</width>
      <height>32</height>
     </rect>
    </property>
    <property name="text">
     <string>Help</string>
    </property>
   </widget>
  </widget>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections/>
</ui>
